### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by yosefe** [comment]
@alex-mikheev @brminich pls take a look

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/872/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3340/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2834/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

File: src/uct/ib/ud/base/ud_ep.h [hunk 1]
```diff
@@ -190,7 +190,10 @@ enum {
     /* debug flags */
     UCT_UD_EP_FLAG_PRIVATE           = UCS_BIT(1),
     UCT_UD_EP_FLAG_CREQ_RCVD         = UCS_BIT(2),
-    UCT_UD_EP_FLAG_CREP_RCVD         = UCS_BIT(3)
+    UCT_UD_EP_FLAG_CREP_RCVD         = UCS_BIT(3),
+    UCT_UD_EP_FLAG_CREQ_SENT         = UCS_BIT(4),
+    UCT_UD_EP_FLAG_CREP_SENT         = UCS_BIT(5),
+    UCT_UD_EP_FLAG_CREQ_NOTSENT      = UCS_BIT(6)
```
Comment (brminich) [inline src/uct/ib/ud/base/ud_ep.h]: minor: seems like it is not needed.  it is enough to check whether UCT_UD_EP_FLAG_CREQ_SENT is set or not.

File: test/gtest/uct/test_ud.cc [hunk 2]
```diff
@@ -120,14 +126,18 @@ class test_ud : public ud_base_test {
...
 
+    unsigned no_creq_cnt(uct_ud_ep_t *ep) {
+        return (ep->flags & UCT_UD_EP_FLAG_CREQ_NOTSENT) ? 1 : 0;
+    }
+
     void validate_recv(uct_ud_ep_t *ep, unsigned value,
                        double timeout_sec=TEST_UD_TIMEOUT_IN_SEC) {
         ucs_time_t timeout = ucs_get_time() + ucs_time_from_sec(timeout_sec);
-        while ((ucs_frag_list_sn(&ep->rx.ooo_pkts) < value) &&
+        while ((ucs_frag_list_sn(&ep->rx.ooo_pkts) < value - no_creq_cnt(ep)) &&
```
Comment (brminich) [inline test/gtest/uct/test_ud.cc]: Do we want to check whether CREP arrived or not? If yes, maybe it is better to check UCT_UD_EP_FLAG_CREP_RCVD flag?

File: src/uct/api/tl.h [hunk 3]
```diff
@@ -48,6 +48,8 @@ typedef struct uct_iface_ops {
 
     void         (*iface_wakeup_close)(uct_wakeup_h wakeup);
 
+    int (*iface_progress_set)(uct_iface_h iface, int enable, int disable);
```
Comment (brminich) [inline src/uct/api/tl.h]: why is this needed?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2835/ for details (Mellanox internal link).

File: test/gtest/uct/test_ud.cc [hunk 4]
```diff
@@ -120,14 +126,18 @@ class test_ud : public ud_base_test {
...
                (ucs_get_time() < timeout)) {
             progress();
         }
-        EXPECT_EQ(value, ucs_frag_list_sn(&ep->rx.ooo_pkts));
+        EXPECT_EQ(value - no_creq_cnt(ep), ucs_frag_list_sn(&ep->rx.ooo_pkts));
```
Comment (alex-mikheev) [inline test/gtest/uct/test_ud.cc]: it is better not to convert boolean into psn. Things will break if initial psn is changed

File: src/uct/ib/ud/base/ud_ep.h (hunk 1 above)
Comment (alex-mikheev) [inline src/uct/ib/ud/base/ud_ep.h]: A notsent flag description is needed. The name by itself is not really descriptive.

File: test/gtest/uct/test_ud.cc (hunk 4 above)
Comment (yosefe) [inline test/gtest/uct/test_ud.cc]: the function returns number of CREQs that were not sent

File: src/uct/ib/ud/base/ud_ep.h (hunk 1 above)
Comment (yosefe) [inline src/uct/ib/ud/base/ud_ep.h]: i want to make sure that CREQ was not sent for a good reason (e.g was not sent intentionally), rather than just "not sent *yet*

File: test/gtest/uct/test_ud.cc (hunk 2 above)
Comment (yosefe) [inline test/gtest/uct/test_ud.cc]: this test expects CREP to arrive, in case we actually sent the CREQ (actually, we count received CREPs rather than xmitted CREQs here)

File: src/uct/api/tl.h (hunk 3 above)
Comment (yosefe) [inline src/uct/api/tl.h]: error, will remove

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3343/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/875/ for details.

File: test/gtest/uct/test_ud.cc (hunk 2 above)
Comment (brminich) [inline test/gtest/uct/test_ud.cc]: ic, right CREP delivery (not only payload) should be checked by test as well.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2838/ for details (Mellanox internal link).

**Comment by brminich** [comment]
:+1:
